frontend/src/app/modules/interventions/list/table/table.component.ts

```diff
@@ -1,44 +1,63 @@
-import { Component, Input, ViewChild, OnChanges, AfterViewInit } from '@angular/core';
+import { AfterViewInit, Component, OnInit, ViewChild } from '@angular/core';
 import { Router } from '@angular/router';
-import { MatTable, MatTableDataSource, MatPaginator } from '@angular/material';
+import { MatPaginator, MatSort, MatTable } from '@angular/material';
 
-import { Intervention } from '@shared/domain/intervention.model';
+import { Intervention, SortDirection, InterventionsFilter } from '@shared/domain/intervention.model';
+import { InterventionsDatasource } from './interventions.datasource';
+import { InterventionsService } from '../../interventions.service';
+import { tap } from 'rxjs/operators';
+import { merge } from 'rxjs';
+
+const defaultSearchParams: InterventionsFilter = {
+  pageSize: 10,
+  page: 1,
+  sortBy: 'creationDate',
+  sortDirection: SortDirection.Descending,
+};
 
 @Component({
   selector: 'app-interventions-table',
   templateUrl: './table.component.html',
   styleUrls: ['./table.component.scss'],
 })
-export class TableComponent implements OnChanges, AfterViewInit {
+export class TableComponent implements AfterViewInit, OnInit {
   displayedColumns = ['position', 'name', 'status', 'phone', 'date', 'description'];
-  dataSource: MatTableDataSource<Intervention>;
-
-  @Input() interventions: Intervention[] = [];
+  dataSource: InterventionsDatasource;
 
   @ViewChild(MatPaginator, { static: true }) paginator: MatPaginator;
   @ViewChild(MatTable, { static: true }) table: MatTable<any>;
+  @ViewChild(MatSort, { static: true }) sort: MatSort;
 
-  constructor(private router: Router) {}
-
-  ngOnChanges(changes: any) {
-    this.checkInterventions(changes.interventions.currentValue, changes.interventions.previousValue);
-  }
-
-  private checkInterventions(currentInterventions?: Intervention[], previousInterventions?: Intervention[]) {
-    if (currentInterventions !== previousInterventions) {
-      this.dataSource = new MatTableDataSource<Intervention>(currentInterventions);
-      this.dataSource.paginator = this.paginator;
-    }
-  }
+  constructor(private router: Router, private interventionsService: InterventionsService) {}
 
   showDetails(intervention: Intervention) {
     this.router.navigate(['interwencje', intervention.id]);
   }
 
+  ngOnInit() {
+    this.dataSource = new InterventionsDatasource(this.interventionsService);
+    this.dataSource.loadInterventions(defaultSearchParams);
+  }
+
   ngAfterViewInit() {
     // TODO: Workaround for https://github.com/angular/components/issues/8057
     // Remove when fixed. Debounces original ngOnDestroy way after route animation finishes.
     const tableOnDestroy = this.table.ngOnDestroy.bind(this.table);
     this.table.ngOnDestroy = () => setTimeout(tableOnDestroy, 1000);
+    this.sort.sortChange.subscribe(() => (this.paginator.pageIndex = 0));
```

Comment:
Extending your idea, let's first create a private method to encapsulate the logic, and in this private method we do:
```
private subscribeToSortAndPageChanges() {
    const sort$ = this.sort.sortChange.pipe(
      tap(() => (this.paginator.pageIndex = 0)),
      startWith({active: 'date', direction: 'desc'})
    );
    const page$ = this.paginator.page.pipe(
      startWith({
        length: 100,
        pageIndex: 1,
        pageSize: 10,
        previousPageIndex: 0
      })
    );
    const toFilters = ([sort, pageEvent]): InterventionsFilter => ({
      page: pageEvent.pageIndex + 1,  // API first page is 1, paginator 0
      pageSize: pageEvent.pageSize,
      sortDirection: sort.direction === 'asc' ? SortDirection.Ascending : SortDirection.Descending,
    });

    combineLatest([sort$, page$])
      .pipe(
        map(toFilters),
        switchMap(filters => this.dataSource.loadInterventions$(filters))
      )
      .subscribe();
  }
```

Reply:
I know this can be overwhelming at first, but this is a fully reactive approach :) 
Try to analyze it first by yourself, but we can also set-up a call where I can explain to you line-by-line why this works and why it is better :)

Reply:
Nice, thanks, this looks better! 🙂